### ISSUE TITLE
Problem omega decrease, but group_sparsity not increase

### ISSUE THREAD

**Issue by C0NGTRI123** [tianyic/only_train_once#43]
I catch this case when I apply prune in task face recognition in repo insightface, I only prune backbone, head use to softmax class. I try to print result, omega decrease but group_sparsity not increase, loss converge (loss calculate: get embeeding of backbone - embedding original of image). I don't know optimizer of OTO converge or my loss correct when apply OTO. I try to increase epoch, but it still catch this case. 

Here is architure 
[IResNet_zig.gv.pdf](https://github.com/tianyic/only_train_once/files/13655623/IResNet_zig.gv.pdf)

**Comment by tianyic** [comment]
Thanks for the new issue, which is actually a common complain regarding the public version.

In our internal version to be released soon, the group sparsity could be precisely controlled. Therefore, I would recommend retrying upon the coming next release.

For a hotfix, you could try to improve lambda or other hyper-parameters to see if the issues could be resolved. 
https://github.com/tianyic/only_train_once/issues/34#issuecomment-1693571894

**Comment by C0NGTRI123** [comment]
I try to change hyper-parameters, they decrease omega fast and group_sparsity increase. But I check onnx_model, I see my channel not decrease, so my model keep model_size although group_sparsity is 0.7. I don't know group_sparsity meaning, I think they group channel remove zero-group. But in my weight, zero_group is so less to have and pruned so my model keep size

**Comment by tianyic** [comment]
It might be due to some node groups are not marked as prunable or some tensor alignment between torch and onnx format. 

I would recommend to retry over the coming new version, which is way better than the public version (having not updated and maintained for a while).

[RDB_Pruning_Dependency_Graph](https://github.com/tianyic/only_train_once/files/13663895/simpleSR_zig.gv.pdf)

**Comment by C0NGTRI123** [comment]
> It might be due to some node groups are not marked as prunable or some tensor alignment between torch and onnx format.
> 
> I would recommend to retry over the coming new version, which is way better than the public version (having not updated and maintained for a while).
> 
> [RDB_Pruning_Dependency_Graph](https://github.com/tianyic/only_train_once/files/13663895/simpleSR_zig.gv.pdf)

I have another question. This framework support for DDP training?

**Comment by tianyic** [comment]
Yes, it naturally supports DDP, and we have plans to further support model parallelism in 2024.
